syntax = "proto3";

service IoTService {
  rpc AddNewUser (Credentials) returns (AddNewUserReply) {}

  rpc GetAccessToken (Credentials) returns (Token) {}

  rpc GetRegions (GetRegionsRequest) returns (RegionsReply) {}
  rpc AddRegion (AddRegionRequest) returns (AddRegionReply) {}
  rpc RemoveRegion (RemoveRegionRequest) returns (RemoveRegionReply) {}

  rpc GetLastRoute (GetLastRouteRequest) returns (RouteReply) {}
  rpc SetRoute (SetRouteRequest) returns (SetRouteReply) {}

  rpc SayTemperature (TemperatureRequest) returns (TemperatureReply) {}
  rpc BlinkLed (LedRequest) returns (LedReply) {}
  rpc SayLuminosity (LuminosityRequest) returns (LuminosityReply) {}
}

message Credentials {
  string login = 1;
  string password = 2;
}

message AddNewUserReply {
  string status = 1;
}

message Token {
  string status = 1;
  string token = 2;
}

message TemperatureRequest {
  string sensorName = 1;
  string accessToken = 2;
}

message TemperatureReply {
  string status = 1;
  string temperature = 2;
}

message LedRequest {
  uint32 state = 1;
  string sensorName = 2;
  string accessToken = 3;
}

message LedReply {
  string status = 1;
  map<string, uint32> ledstate = 2;
}

message LuminosityRequest {
  string sensorName = 1;
  string accessToken = 2;
}

message LuminosityReply {
  string status = 1;
  string luminosity = 2;
}